### PR TITLE
Fixed Null pointer exception in OtpManager and RestRequestFactory

### PR DESCRIPTION
**Pull request by vipuldhurve02**
- response can be null in line 135 and 140 (OtpManager)
- headers.getContentType() can be null in line 88 (RestRequestFactory)

### CODE REVIEW

File: authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/factory/RestRequestFactory.java [hunk 1]
```diff
@@ -84,18 +85,20 @@ public RestRequestDTO buildRequest(RestServicesConstants restService, Object req
...
-	    } else {
-		if (requestBody instanceof MultiValueMap) {
-		    request.setRequestBody(requestBody);
-		} else {
-		    throw new IDDataValidationException(
-			    IdAuthenticationErrorConstants.INVALID_INPUT_PARAMETER.getErrorCode(),
-			    String.format(IdAuthenticationErrorConstants.INVALID_INPUT_PARAMETER.getErrorMessage(),
-				    "requestBody"));
+		if(headers.getContentType()!=null){
+			if (!Objects.requireNonNull(headers.getContentType()).includes(MediaType.MULTIPART_FORM_DATA)) {
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/factory/RestRequestFactory.java]: `headers.getContentType()!=null` needs to be added only for the first if condition. The else condition is not related to the content type, so it should not be a new common outer if condition having the null check.

File: authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/OTPManager.java [hunk 2]
```diff
@@ -135,12 +132,17 @@ private String generateOTP(String uin) throws IdAuthUncheckedException {
 			RestRequestDTO restRequest = restRequestFactory.buildRequest(RestServicesConstants.OTP_GENERATE_SERVICE,
 					reqWrapper, ResponseWrapper.class);
 			ResponseWrapper<Map<String, String>> response = restHelper.requestSync(restRequest);
-			if (response != null && response.getResponse().get("status").equals(USER_BLOCKED)) {
+			if ( response!=null && response.getResponse()!=null && response.getResponse().get("status").equals(USER_BLOCKED)) {
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/OTPManager.java]: add a null check for `response.getResponse().get("status")` also.

File: authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/OTPManager.java [hunk 3]
```diff
@@ -135,12 +132,17 @@ private String generateOTP(String uin) throws IdAuthUncheckedException {
...
 					reqWrapper, ResponseWrapper.class);
 			ResponseWrapper<Map<String, String>> response = restHelper.requestSync(restRequest);
-			if (response != null && response.getResponse().get("status").equals(USER_BLOCKED)) {
+			if ( response!=null && response.getResponse()!=null && response.getResponse().get("status").equals(USER_BLOCKED)) {
 				logger.error(IdAuthCommonConstants.SESSION_ID, this.getClass().getSimpleName(),
 						IdAuthenticationErrorConstants.BLOCKED_OTP_VALIDATE.getErrorCode(), USER_BLOCKED);
 				throw new IdAuthUncheckedException(IdAuthenticationErrorConstants.BLOCKED_OTP_VALIDATE);
 			}
-			return response.getResponse().get("otp");
+			if(response !=null){
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/OTPManager.java]: Add null check for `response.getResponse()`

File: authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/OTPManager.java [hunk 4]
```diff
@@ -135,12 +132,17 @@ private String generateOTP(String uin) throws IdAuthUncheckedException {
...
+				return response.getResponse().get("otp");
+			}else{
+				throw new IdAuthUncheckedException(IdAuthenticationErrorConstants.UNABLE_TO_PROCESS);
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/OTPManager.java]: Throw `IdAuthenticationErrorConstants.OTP_GENERATION_FAILED` here

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/OTPManager.java (hunk 2 above)
Comment (vipuldhurve02) [reply authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/integration/OTPManager.java]: Added check for response.getResponse().get("status") also.

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**